experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -0,0 +1,248 @@
+        restart_dir,
+        restart_t,
+        use_coupler_diagnostics,
+        calendar_dt,
+        evolving_ocean,
+        mono_surface,
+        turb_flux_partition,
+        energy_check,
+        conservation_softfail,
+        output_dir_root,
+        generate_ci_plots,
+        land_domain_type,
+        land_albedo_type,
+        land_temperature_anomaly,
+        use_land_diagnostics,
+    )
+end
+
+"""
+    get_atmos_args(atmos_config_dict)
+
+Extract the necessary arguments from the atmosphere configuration dictionary.
+
+# Arguments
+- `atmos_config_dict`: A dictionary mapping atmosphere configuration keys to the specified settings
+
+# Returns
+- All arguments needed for the atmosphere simulation
+"""
+function get_atmos_args(atmos_config_dict)
+    dt_rad = atmos_config_dict["dt_rad"]
+    output_default_diagnostics = atmos_config_dict["output_default_diagnostics"]
+
+    return (; dt_rad, output_default_diagnostics)
+end
+
+
+### Helper functions used in argument parsing ###
+
+"""
+    get_diag_period()
+
+Determine the frequency at which to average and output diagnostics based on the
+simulation start and end times.
+
+The default periods are:
+- 1 month for simulations longer than 90 days
+- 10 days for simulations longer than 30 days
+- 1 day for simulations longer than 1 day
+- 1 hour for simulations shorter than 1 day
+
+# Arguments
+- `t_start`: The start time of the simulation
+- `t_end`: The end time of the simulation
+
+# Returns
+- `period`: A String of how often to average and output diagnostics
+- `calendar_dt`: A DateTime interval representing the period
+"""
+function get_diag_period(t_start, t_end)
+    sim_duration = t_end - t_start
+    secs_per_day = 86400
+    if sim_duration >= 90 * secs_per_day
+        # if duration >= 90 days, take monthly means
+        period = "1months"
+        calendar_dt = Dates.Month(1)
+    elseif sim_duration >= 30 * secs_per_day
+        # if duration >= 30 days, take means over 10 days
+        period = "10days"
+        calendar_dt = Dates.Day(10)
+    elseif sim_duration >= secs_per_day
+        # if duration >= 1 day, take daily means
+        period = "1days"
+        calendar_dt = Dates.Day(1)
+    else
+        # if duration < 1 day, take hourly means
+        period = "1hours"
+        calendar_dt = Dates.Hour(1)
+    end
+    return (period, calendar_dt)
+end
+
+"""
+    parse_component_dts(config_dict)
+
+Check which timesteps are specified in the config file, and use them to choose
+the correct timestep for each component model.
+If all component timesteps `dt_\$component` are specified in the config file, use those.
+Otherwise, use the generic component timestep `dt` specified in the config file.
+
+# Arguments
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+
+# Returns
+- `component_dt_dict`: A dictionary mapping component model names to their respective timesteps
+"""
+function parse_component_dts(config_dict)
```

Comment:
Maybe we don't need to modify the `config_dict` here. Removing unused dts isn't really needed because the config dict is only used again to create the `atmos_config_dict`.  

If we don't remove unused dts, then the only case we would change the `config_dict` is when `dt_atmos` is specified but not all the other components are. Instead of reading dt_atmos/dt from the `config_dict` in `get_atmos_config_dict`, we could pass in `component_dt_dict`, or just `dt_atmos` to `get_atmos_config_dict`, and then overwrite `dt_atmos` in `atmos_config_dict` after merging with `dt_config`.

Reply:
I thought about this a bit, and here's what I came up with

> If we don't remove unused dts, then the only case we would change the config_dict is when dt_atmos is specified but not all the other components are.

I agree, but we should avoid thinking about `dt_atmos` as a special case. For example, we might want to set up land with a config dict in the future, in which case all the same things would apply to `dt_land`.

> Instead of reading dt_atmos/dt from the config_dict in get_atmos_config_dict, we could pass in component_dt_dict, or just dt_atmos to get_atmos_config_dict, and then overwrite dt_atmos in atmos_config_dict after merging with dt_config.

We could do this, but we should avoid treating certain arguments differently from the others. Right now the only things we pass into `get_atmos_config_dict` are `coupler_dict` and `job_id` (which is special because we read it in from the command line rather than from the dict)

To me, it seems like the best options are:
1. Add `component_dt_dict` to the `config_dict` after we finish choosing which dts to use. Then we don't have to remove the unused keys from the dict, and it will be clear to the component models that they should use whichever dt appears in the `component_dt_dict`. However, I don't like that this solution changes the structure of keys in the config dictionary at different points (e.g. when parsing dts for the coupler vs for the atmos/land), and I don't like that this keeps keys in the config dicts that are unused
2. Clean up the dts when we parse the config file (as is done now). As I think about it more, I think it's okay to have a "cleanup" phase when parsing the config. We could extend this to check consistency more generally, e.g. that EDMF parameters are not in the config if EDMF is not being used.

Right now I prefer 2, but let me know what you think. We can also discuss in person

Reply:
2 seems good to me, especially if there are more keys that could potentially be removed. I think adding new keys into `config_dict` (what would happen in option 1 if component_dts are not specified) could be confusing later on, but removing keys is probably not as confusing.